src/js/events/MesosSummaryActions.js

```diff
@@ -1,58 +1,95 @@
-import ActionTypes from '../constants/ActionTypes';
+import {
+  REQUEST_MESOS_HISTORY_SUCCESS,
+  REQUEST_MESOS_HISTORY_ONGOING,
+  REQUEST_MESOS_SUMMARY_SUCCESS,
+  REQUEST_MESOS_SUMMARY_ERROR,
+  REQUEST_MESOS_SUMMARY_ONGOING
+} from '../constants/ActionTypes';
+
 var AppDispatcher = require('./AppDispatcher');
 var Config = require('../config/Config');
 var RequestUtil = require('../utils/RequestUtil');
 var _historyServiceOnline = true;
 
-function getStateUrl(timeScale) {
-  timeScale = timeScale || 'last';
-  if (_historyServiceOnline) {
-    return `${Config.historyServer}/dcos-history-service/history/${timeScale}`;
-  } else {
-    return `${Config.rootUrl}/mesos/master/state-summary`;
+function testHistoryOnline() {
+  RequestUtil.json({
+    url: `${Config.historyServer}/dcos-history-service/history/last`,
+    success: function () {
+      _historyServiceOnline = true;
+    },
+    error: function () {
+      setTimeout(testHistoryOnline, Config.testHistoryInterval);
+    }
+  });
+}
+
+function getHistory(resolve, reject, timeScale = 'last') {
```

Comment:
Function name could be better, agreed. If you look at `MesosSummaryStore`, it only cares if it gets an array of states `.._HISTORY_..` or a single state `.._SUMMARY_..` irrespective of the API that provided it. We could perhaps rename the constants to avoid confusion.